src/Ide/PluginUtils.hs

```diff
@@ -0,0 +1,103 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Ide.PluginUtils where
+
+import qualified Data.Text as T
+import           Data.Maybe
+import           Data.Algorithm.DiffOutput
+import           Data.Algorithm.Diff
+import qualified Data.HashMap.Strict as H
+import           Language.Haskell.LSP.Types.Capabilities
+import qualified Language.Haskell.LSP.Types            as J
+import           Language.Haskell.LSP.Types
+
+data WithDeletions = IncludeDeletions | SkipDeletions
+  deriving Eq
+
+-- | Generate a 'WorkspaceEdit' value from a pair of source Text
+diffText :: ClientCapabilities -> (Uri,T.Text) -> T.Text -> WithDeletions -> WorkspaceEdit
+diffText clientCaps old new withDeletions =
+  let
+    supports = clientSupportsDocumentChanges clientCaps
+  in diffText' supports old new withDeletions
```

Comment:
Actually unneeded, but might spare others to move the code? Or should this be reverted?

Reply:
Keep it, we can clean up after hie is deprecated and hls is stable.